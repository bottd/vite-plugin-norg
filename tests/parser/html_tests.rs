use insta::assert_yaml_snapshot;
use rstest::rstest;
use std::fs;
use vite_plugin_norg_parser::{extract_metadata, extract_toc, transform};

#[test]
fn test_convert_nodes() {
    use rust_norg::NorgAST::Heading;

    let node = Heading {
        level: 1,
        title: vec![],
        content: vec![],
        extensions: Default::default(),
    };

    let ast = vec![node];
    let html = transform(&ast);
    let toc = extract_toc(&ast);

    assert_yaml_snapshot!((html, toc));
}

#[rstest]
#[case::basic("tests/fixtures/basic.norg")]
#[case::code_blocks("tests/fixtures/code-blocks.norg")]
#[case::headings("tests/fixtures/headings.norg")]
#[case::images("tests/fixtures/images.norg")]
#[case::links("tests/fixtures/links.norg")]
fn test_norg_fixture_files(#[case] fixture_path: &str) {
    let content = fs::read_to_string(fixture_path)
        .unwrap_or_else(|_| panic!("Failed to read {}", fixture_path));
    let ast = rust_norg::parse_tree(&content)
        .unwrap_or_else(|_| panic!("Failed to parse {}", fixture_path));

    let html = transform(&ast);
    let toc = extract_toc(&ast);

    let metadata = extract_metadata(&ast);
    assert_yaml_snapshot!(fixture_path, (html, toc, metadata));
}
