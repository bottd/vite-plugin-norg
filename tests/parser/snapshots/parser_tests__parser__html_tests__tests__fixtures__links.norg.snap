---
source: tests/parser/html_tests.rs
expression: "(html, toc, metadata)"
---
- "<h1 id=\"link-types\">Link Types</h1>\n<h2 id=\"url-links\">URL Links</h2>\n<p><a href=\"https://example.com\" target=\"_blank\">https://example.com</a> <a href=\"https://github.com/nvim-neorg\" target=\"_blank\">Neorg GitHub</a></p><h2 id=\"file-links\">File Links</h2><h2 id=\"heading-links\">Heading Links</h2>\n<p><a href=\"#token-text-main-token-whitespace-token-text-heading\">Token(Text(&quot;Main&quot;))Token(Whitespace)Token(Text(&quot;Heading&quot;))</a> <a href=\"#token-text-sub-token-whitespace-token-text-heading\">Token(Text(&quot;Sub&quot;))Token(Whitespace)Token(Text(&quot;Heading&quot;))</a> <a href=\"#token-text-introduction\">Custom link text</a></p><h2 id=\"mixed-content\">Mixed Content</h2>\n<p>This paragraph has a <a href=\"https://rust-lang.org\" target=\"_blank\">Rust website</a> and .</p>"
- - level: 1
    title: Link Types
    id: link-types
  - level: 2
    title: URL Links
    id: url-links
  - level: 2
    title: File Links
    id: file-links
  - level: 2
    title: Heading Links
    id: heading-links
  - level: 2
    title: Mixed Content
    id: mixed-content
- author: Drake Bott
  title: Link Test
